examples/indexing/fancy_indexing.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+from numpy.random import randint
+
+# %out = tensor.gather %input[%indices] gather_along_dimension(1) :
+#   tensor<2x3x4xf32>[tensor<5x6x 1xindex>] -> tensor<5x6x2x1x4xf32>
```

Comment:
Yes the RFC mostly has `x1x` i.e., "rank-preserving" semantics but also mentions "rank-reducing" semantics. All the examples in this file (which, btw, is just me wrapping my head around the semantics discussed in the RFC and in upstream `TensorOps.td`) are rank-reducing (because `numpy` just naturally corresponds). For the actual implementation (wrt to `Indexing` Python bindings) there will be a way to specify `rank_reducing=True/False`.